frontend/src/views/positions/positions-list-toolbar.js

```diff
@@ -0,0 +1,148 @@
+import React from "react";
+import { formatColumnName } from "../../libs/utils";
+import { ConnectedAddPositionDialog } from "./add-position-dialog";
+import {
+    Badge,
+    Button,
+    ButtonToolbar,
+    ToggleButton,
+    ToggleButtonGroup,
+} from "react-bootstrap";
+
+const BLACKLIST = ["instructor_preferences"]; // Any fields here will not be accessible in the column selector.
+
+/**
+ * Builds an array (with no duplicates) of all column headings which appear at least once in the positions data from the database
+ * All headings which are present in the blacklist (defined above) are removed from this list
+ *
+ * @param {*} positions the positions data which was returned from a GET request to the admin/positions endpoint.
+ * @returns an array with no duplicates containing headings in the format [a-z][a-z]*_[a-z]*
+ */
+function getFetchedColumnHeadings(positions) {
+    return [
+        ...new Set(
+            positions
+                .reduce(
+                    (acc, position) => [...acc, ...Object.keys(position)],
+                    []
+                )
+                .filter((heading) => !BLACKLIST.includes(heading))
+        ),
+    ];
+}
```

Comment:
I don't like magic...Let's list these out by hand. (The order of columns is also very important and must be specified by hand anyways.)